app/models/pglogical_subscription.rb

```diff
@@ -1,36 +1,75 @@
 require 'pg/dsn_parser'
 require 'pg/logical_replication'
+require 'query_relation'
 
-class PglogicalSubscription < ActsAsArModel
-  set_columns_hash(
-    :id                   => :string,
-    :status               => :string,
-    :dbname               => :string,
-    :host                 => :string,
-    :user                 => :string,
-    :password             => :string,
-    :port                 => :integer,
-    :provider_region      => :integer,
-    :provider_region_name => :string
-  )
-
-  def self.find(*args)
-    case args.first
-    when :all then find_all
-    when :first, :last then find_one(args.first)
-    else find_id(args.first)
+class PglogicalSubscription
+  include Vmdb::Logging
+  include ActiveModel::Model
```

Comment:
Without `ActiveModel::Model` you cannot do `PgLogicalSubscription.new` with hash arguments.

Reply:
Can we just define initialize to handle the correct arguments rather than including a whole module? Or is there something wrong with that approach?

Reply:
Well, not only does it mean I don't have to write that code (I'm lazy), it also gives us other potentially useful features should we choose to use them:

https://guides.rubyonrails.org/active_model_basics.html#model

Reply:
I think I would rather keep this simple as I'm not convinced all those features would work with a model not backed by a table. And I don't want to add untested capabilities without a good reason.

Reply:
> I think I would rather keep this simple as I'm not convinced all those features would work with a model not backed by a table.

That's the point of `include`-ing `ActiveModel::Model` into a class, though. Otherwise you would just subclass `ActiveRecord::Base`.

> And I don't want to add untested capabilities without a good reason.

I'm not sure what you mean by "untested capabilities". I can add a cross-repo check to make sure nothing else breaks.

Reply:
I mean the other "poetntially useful features" which I'm not convinced would work for a model backed by a stored procedure. This, for what it's worth, is also why I don't think we could subclass `ActiveRecord::Base` as it would expect a valid table name.

The simplicity I'm shooting for here would mean including as few dependencies as possible, not writing as little code as possible.

Reply:
@carbonin Ok, updated to remove `ActiveModel::Model` and just use a custom `initialize` method.

Reply:
:bowing_man: Thanks @djberg96